Packs/qualys/Scripts/QualysCreateIncidentFromReport/QualysCreateIncidentFromReport.py

```diff
@@ -0,0 +1,129 @@
+                sys.exit(0)
+
+            asset_id = get_asset_id_for_ip(ip)
+            if not asset_id:
+                demisto.results({
+                    "Type": entryTypes["error"],
+                    "ContentsFormat": formats["text"],
+                    "Contents": 'No ID was found for asset {0}'.format(str(asset))
+                })
+                sys.exit(0)
+
+            # Get Asset vulnerability list
+            vulnerabilities = argToList(demisto.get(asset, "VULN_INFO_LIST.VULN_INFO"))
+            if not isinstance(vulnerabilities, list):
+                vulnerabilities = [vulnerabilities]
+
+            qids = map(lambda vulnerability: demisto.get(vulnerability, "QID.#text"), vulnerabilities)
+
+            # Get only the QIDs that exists in asset and has severity >= min_severity
+            qids = list(set(qids) & set(qid_severity))
+
+            for qid in qids:
+                # Search for existing open incidents with the same Vendor ID and Asset ID.
+                # Will open a new incident only if such an incident not exists.
+                resp = demisto.executeCommand(
+                    "getIncidents",
+                    {"query": "vendorid: {0} and assetid: {1} and --status:Closed".format(qid, asset_id)})
+                if isError(resp[0]):
+                    demisto.results(resp)
+                    sys.exit(0)
+
+                incident_number = demisto.get(resp[0], "Contents.total")
+
+                try:
+                    incident_number = int(incident_number)
+                except Exception:
+                    demisto.results({
+                        "Type": entryTypes["error"],
+                        "ContentsFormat": formats["text"],
+                        "Contents": 'Error while searching the incident repository'
+                    })
+                    sys.exit(0)
+
+                if incident_number == 0:
+                    # Create incident
+                    demisto.executeCommand("createNewIncident", {
+                        "name": "Vulnerability - Asset {0} QID {1} - {2}".format(asset_id, qid, generation_date),
+                        "vendorid": str(qid),
+                        "type": incident_type,
+                        "assetid": str(asset_id)
+                    })
+
+        demisto.results("Done.")
+    else:
+        demisto.results({
+            "Type": entryTypes["error"],
+            "ContentsFormat": formats["text"],
+            "Contents": 'No data could be read.'
+        })
+
+
+def get_asset_id_for_ip(ip):
+    resp = demisto.executeCommand("qualys-host-list", {"ips": ip})
+    if isError(resp[0]):
+        demisto.results(resp)
+        sys.exit(0)
+
+    try:
+        asset_id = demisto.get(resp[0], 'Contents.HOST_LIST_OUTPUT.RESPONSE.HOST_LIST.HOST.ID')
+    except TypeError:  # for v2
+        asset_id = demisto.get(json.loads(xml2json(resp[0]['Contents'])), 'HOST_LIST_OUTPUT.RESPONSE.HOST_LIST.HOST.ID')
```

Comment:
This is ok (no need to change), but FYI we have in CSP the `execute_command()` method, which wraps all this more elegantly.